src/cmd/linuxkit/pkglib/build.go

```diff
@@ -227,45 +248,57 @@ func (p Pkg) Build(bos ...BuildOpt) error {
 	if !skipBuild {
 		fmt.Fprintf(writer, "building %s\n", ref)
 		var (
-			args  []string
+			imageBuildOpts = types.ImageBuildOptions{
+				Labels:    map[string]string{},
+				BuildArgs: map[string]*string{},
+			}
 			descs []registry.Descriptor
 		)
 
+		// args that we use:
+		//   labels map[string]string
+		//   network string
+		//   build-arg []string
+
 		if p.git != nil && p.gitRepo != "" {
-			args = append(args, "--label", "org.opencontainers.image.source="+p.gitRepo)
+			imageBuildOpts.Labels["org.opencontainers.image.source"] = p.gitRepo
 		}
 		if p.git != nil && !p.dirty {
 			commit, err := p.git.commitHash("HEAD")
 			if err != nil {
 				return err
 			}
-			args = append(args, "--label", "org.opencontainers.image.revision="+commit)
+			imageBuildOpts.Labels["org.opencontainers.image.revision"] = commit
 		}
 
 		if !p.network {
-			args = append(args, "--network=none")
+			imageBuildOpts.NetworkMode = "none"
 		}
 
 		if p.config != nil {
 			b, err := json.Marshal(*p.config)
 			if err != nil {
 				return err
 			}
-			args = append(args, "--label=org.mobyproject.config="+string(b))
+			imageBuildOpts.Labels["org.mobyproject.config"] = string(b)
 		}
 
-		args = append(args, "--label=org.mobyproject.linuxkit.version="+version.Version)
-		args = append(args, "--label=org.mobyproject.linuxkit.revision="+version.GitCommit)
+		imageBuildOpts.Labels["org.mobyproject.linuxkit.version"] = version.GitCommit
```

Comment:
Should it be `version.Version` as before?

Reply:
Now how did we miss that? Nice catch @giggsoff ; fixing.